src/main/java/com/documaster/rms/noark/ws/serialization/CustomObjectMapper.java

```diff
@@ -19,12 +19,11 @@ public enum CustomObjectMapper {
 	CustomObjectMapper() {
 
 		ObjectMapper mapper = new ObjectMapper();
+		mapper.registerModule(new Jdk8Module());
 		mapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
 		mapper.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE);
 		mapper.setVisibility(PropertyAccessor.GETTER, JsonAutoDetect.Visibility.ANY);
 		mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-		mapper.setDefaultPropertyInclusion(
-				JsonInclude.Value.construct(JsonInclude.Include.NON_NULL, JsonInclude.Include.ALWAYS));
```

Comment:
Another issue that I hit is trying to set the saksstatus to A without specifying avsluttet* fields. This leads to an error when it shouldn't because our backend would set reasonable defaults for them.

Reply:
Maybe a solution to this would be to make all fields optional 😱

Reply:
I started implementing the new approach for serialization first for the non-mandatory code list fields. With the new approach the fields won't be declared optional, but we will keep the optional getter. We will also need to add a boolean flag for when the field was set intentionally.

Among the reasons for the switch to the new implementation is that we broke the reflection logic in `NoarkObjectMappings#gatherCodeValueFields` which was not able to properly enumerate the (previously) optional codelist fields. As far as I checked there was no way to obtain the generic type of the optional field at runtime.